File: Solutions/0347-top-k-frequent-elements/solution.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> cnt;
        for(auto num: nums) cnt[num]++;
        priority_queue<pair<int, int>> pq;
        for(auto el: cnt) pq.push({el.second, el.first});
        vector<int> ans;
        while(k--) {
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};

File: Solutions/0347-top-k-frequent-elements/solution.ts
function topKFrequent(nums: number[], k: number): number[] {
    const mp: Map<number, number> = new Map();
    for(const el of nums) {
        const fr = mp.get(el);
        if(fr) mp.set(el, fr + 1);
        else mp.set(el, 1);
    }
    const arr: [number, number][] = [];
    for(const [num, freq] of mp) {
        arr.push([freq, num]);
    }
    arr.sort((a, b) => b[0] - a[0]);
    const ans: number[] = [];
    for(let i = 0 ; i < k ; i++) {
        ans.push(arr[i][1]);
    }
    return ans;
};

